apps/juicebox_stream/lib/juicebox_stream/youtube/client.ex

```diff
@@ -1,33 +1,52 @@
 defmodule JuiceboxStream.Youtube.Client do
   @youtube_api Application.get_env(:juicebox_stream, :youtube_api)
 
+  @search_params %{
+    part: "snippet",
+    safeSearch: "strict",
+    type: "video",
+    videoCategoryId: "10", # music
+    order: "viewCount",
+    videoEmbeddable: "true"
+  }
+
   def search(query) do
-    with {:ok, response} <- @youtube_api.search(query),
-         parsed = parse_response(response),
-         do: format_response(parsed)
-  end
+    case @youtube_api.search(query, @search_params) do
+      %{"items" => videos} ->
+        results = fetch_video_meta_data(videos)
+        |> merge_responses(videos)
+        |> Enum.map(&format_video/1)
 
-  defp parse_response(%{body: body}) do
-    Poison.Parser.parse!(body)
+        {:ok, results}
+      _ -> {:error, nil}
+    end
   end
 
-  defp format_response(%{"items" => items}) do
-    {:ok, Enum.map(items, &format_video/1)}
+  defp merge_responses(videos_meta, videos) do
+    Enum.map(videos, fn(video) ->
+      meta = Enum.find(videos_meta, &(&1["id"] == video["id"]["videoId"]))
+
+      %{
+         video: video,
+         meta: meta
+       }
+    end)
   end
 
-  defp format_response(%{"error" => error}) do
-    {:error, error}
+  defp fetch_video_meta_data(videos) do
+    Enum.map(videos, &(&1["id"]["videoId"]))
+    |> @youtube_api.videos(%{ part: "contentDetails" })
```

Comment:
There is no whitespace around parentheses/brackets most of the time, but here there is.